hudi-hadoop-common/src/test/java/org/apache/hudi/common/table/TestHoodieTableConfig.java

```diff
@@ -510,4 +522,166 @@ void testInferMergeMode(RecordMergeMode inputMergeMode, String inputPayloadClass
+            5, COMMIT_TIME_ORDERING.name(), null, COMMIT_TIME_BASED_MERGE_STRATEGY_UUID, AWSDmsAvroPayload.class.getName(), null, null, "Op", "D"),
+
+        // Test case: Version 9 table with EventTimeAvroPayload (event time based payload)
+        arguments("Version 9 with EventTimeAvroPayload", null, EventTimeAvroPayload.class.getName(), null, "ts", HoodieTableVersion.NINE,
+            3, EVENT_TIME_ORDERING.name(), null, EVENT_TIME_BASED_MERGE_STRATEGY_UUID, EventTimeAvroPayload.class.getName(), null, null, null, null),
+
+        // Test case: Version 9 table with MySqlDebeziumAvroPayload (event time based payload)
+        arguments("Version 9 with MySqlDebeziumAvroPayload", null, MySqlDebeziumAvroPayload.class.getName(), null, "ts", HoodieTableVersion.NINE,
+            3, EVENT_TIME_ORDERING.name(), null, EVENT_TIME_BASED_MERGE_STRATEGY_UUID, MySqlDebeziumAvroPayload.class.getName(), null, null, null, null)
+    );
+  }
+
+  @ParameterizedTest
+  @MethodSource("argumentsForInferMergingConfigsForVersion9")
+  void testInferMergingConfigsForVersion9(String testName, RecordMergeMode recordMergeMode, String payloadClassName,
+                                          String recordMergeStrategyId, String orderingFieldName, HoodieTableVersion tableVersion,
+                                          int expectedConfigSize, String expectedMergeMode, String expectedPayloadClass,
+                                          String expectedMergeStrategyId, String expectedLegacyPayloadClass,
+                                          String expectedPartialUpdateMode, String expectedDebeziumMarker,
+                                          String expectedDeleteKey, String expectedDeleteMarker) {
+    Map<String, String> configs = HoodieTableConfig.inferMergingConfigsForVersion9(
+        recordMergeMode, payloadClassName, recordMergeStrategyId, orderingFieldName, tableVersion);
+
+    assertEquals(expectedConfigSize, configs.size(), "Config size mismatch for: " + testName);
+    if (expectedMergeMode != null) {
+      assertEquals(expectedMergeMode, configs.get(RECORD_MERGE_MODE.key()),
+          "Merge mode mismatch for: " + testName);
+    }
+    if (expectedPayloadClass != null) {
+      assertEquals(expectedPayloadClass, configs.get(PAYLOAD_CLASS_NAME.key()),
+          "Payload class mismatch for: " + testName);
+    }
+    if (expectedMergeStrategyId != null) {
+      assertEquals(expectedMergeStrategyId, configs.get(RECORD_MERGE_STRATEGY_ID.key()),
+          "Merge strategy ID mismatch for: " + testName);
+    }
+    if (expectedLegacyPayloadClass != null) {
+      assertEquals(expectedLegacyPayloadClass, configs.get(LEGACY_PAYLOAD_CLASS_NAME.key()),
+          "Legacy payload class mismatch for: " + testName);
+    }
+    if (expectedPartialUpdateMode != null) {
+      assertEquals(expectedPartialUpdateMode, configs.get(HoodieTableConfig.PARTIAL_UPDATE_MODE.key()),
+          "Partial update mode mismatch for: " + testName);
```

Comment:
if null, don't we need to validate that the prop is not present in HoodieTableConfig

Reply:
fixing it in next commit